crates/kv-store/src/sstable.rs

```diff
@@ -0,0 +1,1026 @@
+            .field("back_block_idx", &self.back_block_idx)
+            .field("next_first", &self.next_first)
+            .finish()
+    }
+}
+
+impl<'a> SsTableIter<'a> {
+    fn new(table: &'a SsTable) -> Self {
+        Self::new_scan(table, Bound::Unbounded, Bound::Unbounded)
+    }
+
+    pub fn new_scan(table: &'a SsTable, start: Bound<&[u8]>, end: Bound<&[u8]>) -> Self {
+        let (table_idx, mut iter, excluded) = match start {
+            Bound::Included(start) => {
+                let idx = table.find_block_idx(start);
+                let block = table.read_block_cached(idx);
+                let iter = BlockIter::new_seek_to_key(block, start);
+                (idx, iter, None)
+            }
+            Bound::Excluded(start) => {
+                let idx = table.find_block_idx(start);
+                let block = table.read_block_cached(idx);
+                let iter = BlockIter::new_seek_to_key(block, start);
+                (idx, iter, Some(start))
+            }
+            Bound::Unbounded => {
+                let block = table.read_block_cached(0);
+                let iter = BlockIter::new_seek_to_first(block);
+                (0, iter, None)
+            }
+        };
+        let (end_idx, end_iter, end_excluded) = match end {
+            Bound::Included(end) => {
+                let end_idx = table.find_back_block_idx(end);
+                if end_idx == table_idx {
+                    iter.back_to_key(end);
+                    // if the next back is invalid, the next should also be invalid
+                    if !iter.has_next_back() {
+                        iter.next();
+                    }
+                    (end_idx, None, None)
+                } else {
+                    let block = table.read_block_cached(end_idx);
+                    let iter = BlockIter::new_back_to_key(block, end);
+                    (end_idx, Some(iter), None)
+                }
+            }
+            Bound::Excluded(end) => {
+                let end_idx = table.find_back_block_idx(end);
+                if end_idx == table_idx {
+                    iter.back_to_key(end);
+                    // if the next back is invalid, the next should also be invalid
+                    if !iter.has_next_back() {
+                        iter.next();
+                    }
+                    (end_idx, None, Some(end))
+                } else {
+                    let block = table.read_block_cached(end_idx);
+                    let iter = BlockIter::new_back_to_key(block, end);
+                    (end_idx, Some(iter), Some(end))
+                }
+            }
+            Bound::Unbounded => {
+                let end_idx = table.meta.len() - 1;
+                if end_idx == table_idx {
+                    (end_idx, None, None)
+                } else {
+                    let block = table.read_block_cached(end_idx);
+                    let iter = BlockIter::new_seek_to_first(block);
```

Comment:
Should it be `new_seek_to_last`?

Reply:
I rename it to `new`, it means the iter is from the first to the last.